src/uu/tail/src/args.rs

```diff
@@ -302,34 +354,45 @@ impl Settings {
 
         VerificationResult::Ok
     }
+
+    pub fn is_default(&self) -> bool {
+        let default = Self::default();
+        self.max_unchanged_stats == default.max_unchanged_stats
+            && self.sleep_sec == default.sleep_sec
+            && self.follow == default.follow
+            && self.mode == default.mode
+            && self.pid == default.pid
+            && self.retry == default.retry
+            && self.use_polling == default.use_polling
+            && (self.verbose == default.verbose || self.inputs.len() > 1)
+            && self.presume_input_pipe == default.presume_input_pipe
+    }
 }
 
-pub fn arg_iterate<'a>(
-    mut args: impl uucore::Args + 'a,
-) -> UResult<Box<dyn Iterator<Item = OsString> + 'a>> {
-    // argv[0] is always present
-    let first = args.next().unwrap();
-    if let Some(second) = args.next() {
-        if let Some(s) = second.to_str() {
-            match parse::parse_obsolete(s) {
-                Some(Ok(iter)) => Ok(Box::new(vec![first].into_iter().chain(iter).chain(args))),
-                Some(Err(e)) => Err(UUsageError::new(
-                    1,
-                    match e {
-                        parse::ParseError::Syntax => format!("bad argument format: {}", s.quote()),
-                        parse::ParseError::Overflow => format!(
-                            "invalid argument: {} Value too large for defined datatype",
-                            s.quote()
-                        ),
-                    },
-                )),
-                None => Ok(Box::new(vec![first, second].into_iter().chain(args))),
-            }
-        } else {
-            Err(UUsageError::new(1, "bad argument encoding".to_owned()))
+pub fn parse_obsolete(arg: &OsString) -> UResult<Option<parse::ObsoleteArgs>> {
+    match parse::parse_obsolete(arg) {
+        Some(Ok(args)) => Ok(Some(args)),
+        None => Ok(None),
+        Some(Err(e)) => {
+            let arg_str = arg.to_string_lossy();
+            Err(USimpleError::new(
+                1,
+                match e {
+                    parse::ParseError::OutOfRange => format!(
+                        "invalid number: {}: Numerical result out of range",
+                        arg_str.quote()
+                    ),
+                    parse::ParseError::Overflow => format!("invalid number: {}", arg_str.quote()),
+                    parse::ParseError::Context => format!(
+                        "option used in invalid context -- {}",
+                        arg_str.chars().nth(1).unwrap_or_default()
+                    ),
```

Comment:
Wouldn't this be caught by clap already?

Reply:
It would! But clap's error message is different:

```
$ ./target/release/tail -5cz
error: Found argument '-5' which wasn't expected, or isn't valid in this context

  If you tried to supply '-5' as a value rather than a flag, use '-- -5'

Usage: ./target/release/tail [FLAG]... [FILE]...

For more information try '--help'
```

vs

```
$ tail -5cz
tail: option used in invalid context -- 5
```

This is relevant to parse the misc/tail test of GNU (checks for the error output).

Reply:
Yes, but we tend to not be compatible with error messages in general. Instead we can patch the GNU test suite. This is part of what https://github.com/uutils/coreutils/blob/main/util/build-gnu.sh does. I'd say that this PR is ready if it passes all the tests you copied with error messages adapted to clap where necessary.

Reply:
I can remove it if the clap error is generally preferred. But I think in this case we kind of get it "for free". But obviously, when returning None instead of the Error, we will see clap's error instead.

Reply:
Alright, we'll leave it in, but maybe you could add a comment explaining why it's there?